packages/core/src/document.ts

```diff
@@ -258,10 +258,16 @@ class Document extends EventEmitter {
    * @private
    */
   async _applyOpts(opts: DocOpts): Promise<void> {
-    if (!opts.applyOnly) {
+    const anchor = opts.anchor ?? true
+    const publish = opts.publish ?? true
+    const waitForSync = opts.waitForSync ?? true
+    if (anchor) {
       await this.anchor()
-      this._publishTip()
-    } else if (!opts.skipWait) {
+    }
+    if (publish) {
+      await this._publishTip()
+    }
+    if (waitForSync) {
```

Comment:
This introduces a minor behavior change by making waiting for sync the default behavior for all document writes.  Currently we only wait for sync if we skipped anchoring and publishing which seems backwards to me, and is why I changed the behavior here.  Most of the test fallout in this CR adding `{waitForSync:false}` everywhere is because of this behavior change.  Although the real issue with those tests is that they depend on the timing of when anchors are performed, but aren't controlling that themselves.  I filed https://github.com/ceramicnetwork/js-ceramic/issues/528 to fix this in the tests

Reply:
So if you call `change` on a doc will it wait by default? I assume it shouldn't right?

Reply:
So you're thinking we should only wait by default for load operations?  That makes sense, but probably makes sense to encode that default into the change() method rather than have it here.  I'll try that now